### PR TITLE
Register call_requests in background at metatx

### PR DESCRIPTION
**Pull request by kompotkot**
During operations with waggle service we notices few bottlenecks and problems:
- Long lists of call requests force users to split them into multiple lists (<400) and require us to increase the nginx timeout to a high value
- There is no detailed exceptions if few call request IDs already presented in database

With new PR https://github.com/moonstream-to/api/pull/1062 for metatx and this one for waggle next improvements:
- Before pushing call requests - make a check if this call requests represented in database (if `"metatx"` set in request body)
- Track jobs of adding call_request batches with Bugout journals
- Response to user not waiting background jobs

Possible `"metatx"` values:
- `no_metatx` - do not push call_requests to journal, sign it and return back
- `no_check_metatx` - do not check existing call_requests before push to metatx

Additional response fields:
- `job_entry_id` - uuid of entry of job
- `job_entry_url` - full url to job in spire

image

### CODE REVIEW

**Comment by kompotkot** [comment]
@bugout-dev check
- [x] Create journal `metatx-jobs`

File: bugout.go [hunk 1]
```diff
@@ -292,15 +299,15 @@ func (c *BugoutAPIClient) FindGroup(token, groupId string) (Group, int, error) {
 	var requestBodyBytes []byte
 	request, requestErr := http.NewRequest("GET", fmt.Sprintf("%s/group/find?group_id=%s", c.BroodBaseURL, groupId), bytes.NewBuffer(requestBodyBytes))
 	if requestErr != nil {
-		return group, 500, requestErr
+		return group, 0, requestErr
```
Comment (kellan-simiotics) [inline bugout.go:302]: Why are we sending a status code 0 on error?

File: bugout.go [hunk 2]
```diff
@@ -292,15 +299,15 @@ func (c *BugoutAPIClient) FindGroup(token, groupId string) (Group, int, error) {
...
 	}
 	request.Header.Add("Authorization", fmt.Sprintf("Bearer %s", token))
 	request.Header.Add("Accept", "application/json")
 	request.Header.Add("Content-Type", "application/json")
 
 	response, responseErr := c.HTTPClient.Do(request)
 	if responseErr != nil {
-		return group, 500, requestErr
+		return group, 0, requestErr
```
Comment (kellan-simiotics) [inline bugout.go:310]: Same question as above.

**Review by kellan-simiotics — APPROVED** [review]
lg

File: bugout.go (hunk 1 above)
Comment (kompotkot) [reply bugout.go:302]: I want to pass reasonable status codes back to client, so we need to proxy it. When getting 0, it means failed on waggle side, not from other API

**Pull request merged**